Refactored findFigureBehind()

jaykay99992 opened the pull request:
For the first and last findFigureBehind() that takes two paramaters, I replaced them with one method that checks the type of the second paramater , and based on being a Figure or a Colletion of Figures, code will be executed.

For the findFigureBehind() that takes 3 parameters, I rearranged the parameters so that developers do not have to rearrange them when adding a third parameter.
   from
    findFigureBehind( p ,scaleDenominator, figure )
    to
    findFigureBehind(p , figure, scaleDenominator )

    because the other findFigureBehind() are arranged this way (p, figure)

jaykay9999 committed:
Refactored findFigureBehind()

For the first and last findFigureBehind() that takes two paramaters, I replaced them with one method that checks the type of the second paramater , and based on being a Figure or a Colletion of Figures, code will be executed.

    For the findFigureBehind() that takes 3 parameters, I rearranged the parameters so that developers do not have to rearrange them when adding a third parameter.
    from
    findFigureBehind( p ,scaleDenominator, figure )
    to
    findFigureBehind(p , figure, scaleDenominator )

    because the other findFigureBehind() are arranged this way (p, figure)

tony2482012 commented on jhotdraw-core/src/main/java/org/jhotdraw/draw/DefaultDrawing.java:
```diff
@@ -131,11 +131,11 @@ public Figure findFigureExcept(Point2D.Double p, Figure ignore) {
 
   @Override
-  public Figure findFigureBehind(Point2D.Double p, Figure figure) {
+  public Figure findFigureBehind(Point2D.Double p, Figure figure, double scaleDenominator) {
```
Did you refactor all calls to this method to now pass in scaleDenominator?

tony2482012 commented on jhotdraw-core/src/main/java/org/jhotdraw/draw/DefaultDrawing.java:
```diff
@@ -146,35 +146,36 @@ public Figure findFigureBehind(Point2D.Double p, Figure figure) {
   }
 
```
Can we add some java docs and inline comments on these methods to better understand what the code is doing?

tony2482012 commented on jhotdraw-core/src/main/java/org/jhotdraw/draw/DefaultDrawing.java:
```diff
@@ -146,35 +146,36 @@ public Figure findFigureBehind(Point2D.Double p, Figure figure) {
+  public Figure findFigureBehind(Point2D.Double p, Object obj) {
+
+    if (obj instanceof Figure) {
+      boolean isBehind = false;
```
Should this be set to false? We will never hit the if statement on the first loop

jaykay99992 committed:
added javadoc comment for method findFigureBehind

jaykay99992 committed:
Refactored findFIgureBehond in QuadTreeCompositeFigure.java to take 3 parameters

jaykay99992 commented:
@tony2482012 I just refactored getFigureBehind() where it is being called in a different file to also take the new parameter (scaleDenominator) , I also added the javadoc comment. As for the the isBehind being declared "false", that is not an error since it will be manipulated inside the loop. If you have other propositions for the flase declaration, I would gladly change them. is it fine like this? do i merge the code ?

tony2482012 commented:
> @tony2482012 I just refactored getFigureBehind() where it is being called in a different file to also take the new parameter (scaleDenominator) , I also added the javadoc comment. As for the the isBehind being declared "false", that is not an error since it will be manipulated inside the loop. If you have other propositions for the flase declaration, I would gladly change them. is it fine like this? do i merge the code ?

Take a look at my last 2 comments and then I will be good to merge.
I highly recommend adding inline comments to the code. Took me a while to understand what was going on. Having those comments would help all reviewers
